### PR TITLE
Test

### PR DESCRIPTION
**Pull request by GhislainMitahi**
> **IN THIS STAGE I DID**

Add the following validations

> For the User model:

- Name is not blank.

- Posts Counter is an integer greater than or equal to zero.

> For the Post model:

- Title  blank.

- Title is not 250 characters.

- CommentsCounter is not an integer greater than or equal to zero.

- LikesCounter it will not be an integer greater than or equal to zero.

> Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Post.new(title: 'Hello', text: 'This is my first post', commentsCounter: 0, likesCounter: 0, author: user)
+  end
+
+  before { subject.save }
+
+  it 'title should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (anagudelogu) [inline spec/models/post_spec.rb:16]: - [ ] You did a great job working on your model specs! 🚀 However, according to the requirements:

image

You only wrote tests for the presence of the title, please consider writing tests for all validations and methods (`update_posts_counter` and `recent_comments`)! 🤓 🔧

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  it 'post_id should be a string' do
+    subject.post_id = '11123'
+    expect(subject).to_not be_valid
+  end
+
+  it 'user_id should be a string' do
+    subject.user_id = '1123'
+    expect(subject).to_not be_valid
+  end
```
Comment (anagudelogu) [inline spec/models/like_spec.rb:31]: - [ ] The same happens here, you did well validating that the user and post should exist, however, you didn't test your `update_likes_counter` method. Please consider doing it! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @GhislainMitahi, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @GhislainMitahi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] All tests are passing ✔️ 
- [ ] Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**